src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs

```diff
@@ -111,7 +115,8 @@ public long TotalFreeSpace
             get
             {
                 long userBytes, totalBytes, freeBytes;
-                uint oldMode = Interop.Kernel32.SetErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS);
+                uint oldMode;
+                bool success = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out oldMode);
```

Comment:
Is it worth wrapping it in a disposable so you can do a prettier `using` instead?